### PR TITLE
Introduce IOC limit order

### PR DESCRIPTION
**Pull request by beniloew**
In addition, add implementation for Binance

### CODE REVIEW

**Comment by beniloew** [comment]
Im using python, hope that the JS is ok.
Not sure about the async call in binance...

**Comment by beniloew** [comment]
The failure seems not to be related to the change...

**Comment by beniloew** [comment]
Closed and reopened in order to retrigger Travis

**Comment by frosty00** [comment]
In terms of asynchronous javascript I think you need an await inside the body of the function, like so:

        let order = await  this.createOrder (symbol, 'limit', side, amount, price, params);
        return order;

**Comment by beniloew** [comment]
Thanks

**Comment by beniloew** [comment]
Kroitor, is there anything wrong with this pr?

**Comment by kroitor** [comment]
Still not sure if we should have a binance-specific in the base class.
Basically, you did a simple wrapper over:

```JavaScript
await this.createOrder (symbol, 'limit', side, amount, price, {'timeInForce': 'IOC'});
```

And to me it looks like exchange-specific code that belongs to userland, not to the lib itself. The lib gives you createOrder extra params for that. We could, of course, merge it for convenience, but what would you say if I tell you that on a different exchange that same order type is called "Fill or Kill" ? Would it require a createFOKLimitOrder?.. Whatever way I look at it, I don't think we should have an exchange-specific order-type-specific method in the base class.

In other words, I'm afraid, this has to live in userland. Let me know if you disagree. I would not want you to stop contributing because of that, so, I hope for your understanding.

**Comment by beniloew** [comment]
Thanks for your comment.
Well, i disagree :). 
The term IOC is not binance specific, but is known in the finance world. The idea is to have a generic way to put a limit order that is canceled right after the order has been done. I added the binance specific implementation in order to have at least one implementation and not leaving the definition alone. I plan to add other exchanges as well, some that have time in force setting option, e.g. Hitbtc, and some that dont, and i  guess that the implementation will be order and immediate  cancel.

**Comment by kroitor** [comment]
@beniloew ok, thanks for the considerations... I'm for the unification, for sure. 
However, if we project it in future, we will arrive to having this:

- for IOC orders: createIOCLimitOrder
- for FOK orders: createFOKLimitOrder
- for GTC orders: createLimitOrder

So, it kinda breaks the naming here...  The GTC is the default "time in force" type with most exchanges.

From that naming it follows that we should not have a variety of create*XX*LimitOrder methods. Instead, this should be an argument of the createOrder, otherwise we will have an unmanageable zoo of methods.

So, in fact, we may end up adding another argument, or another single method to handle **all** time-in-force types. Like `createTimeInForceLimitOrder (symbol, side, amount, price, timeInForce, params = {})`. A side note on this: we are also avoiding abbreviations as much as possible.

Let me know what you think on the above. Thx!

**Comment by beniloew** [comment]
So, after looking into it it, i think that having a a dedicated function parameter for that is the best.
But, im not sure if the params parameter is good for that, as i see it as a parameter that should be used for exchange specific stuff.
So, i think the best way would be to add another param to the createOrder function, which will default to  
GTC.
Before starting with that, what's your opinion?
Also, FOK cannot be "simulated" on an exchange that does not support time in force setting, but IOC can rather be simulated by creating an order and cancelling it right afterwards.
Will you accept such solution?
Thanks again
